projects/self-creation-portal/src/app/components/resource-holder/resource-holder.component.ts

```diff
@@ -340,23 +284,78 @@ export class ResourceHolderComponent implements OnInit, OnDestroy{
    }
 
   filterButtonClickEvent(event : { label: string }) {
-    this.filters.activeFilterButton = event.label;
-    switch(event.label) {
-      case 'CHANGES_REQUIRED':
-        this.filters.filteredLists = this.lists.filter((item : any) => item.review_status === 'COMMENTS');
-        if(this.filters.filteredLists.length === 0) {
-          this.noResultMessage = "NO_CHANGE_REQUIRED"
-        }
-        break;
-      case 'INPROGRESS':
-        this.filters.filteredLists = this.reviewList.filter((item: any) => item.review_status === 'INPROGRESS');
-        if(this.filters.filteredLists.length === 0) {
-          this.noResultMessage = "NO_INPROGRESS_REVIEW"
-        }
-        break;
+    if(this.filters.activeFilterButton === event.label) {
+      this.filters.activeFilterButton = '';
+      this.filters.filteredLists = this.lists;
+      this.noResultMessage = this.noResultFound;
+    } else {
+      this.filters.activeFilterButton = event.label;
+      switch(event.label) {
+        case 'CHANGES_REQUIRED':
+          this.filters.filteredLists = this.lists.filter((item : any) => item.review_status === 'COMMENTS');
+          if(this.filters.filteredLists.length === 0) {
+            this.noResultMessage = "NO_CHANGE_REQUIRED"
+          }
+          break;
+        case 'INPROGRESS':
+          this.filters.filteredLists = this.lists.filter((item: any) => item.review_status === 'INPROGRESS');
+          if(this.filters.filteredLists.length === 0) {
+            this.noResultMessage = "NO_INPROGRESS_REVIEW"
+          }
+          break;
+      }
     }
   }
 
+  infoIconClickEvent(event: any) {
+    const cardItem = event.item;
+  
+    //to get field data from listapi to map in json
+    const getFieldData = (field: any) => {
+      let value = cardItem[field.name] || '';
+      if (field.name.includes('organization')) {
+        value = cardItem.organization ? cardItem.organization.name : '';
+      } else if (typeof value === 'string' && this.commonService.isISODate(value)) { 
```

Comment:
typeof value === 'string' this can be added in the isISODate function itself know ?